app/modules/xpt2046.c

```diff
@@ -0,0 +1,213 @@
+// Module for xpt2046
+// by Starofall
+// used source from:
+//  - https://github.com/spapadim/XPT2046/
+//  - https://github.com/PaulStoffregen/XPT2046_Touchscreen/
+
+#include "module.h"
+#include "lauxlib.h"
+#include "platform.h"
+#include "c_stdlib.h"
+#include "c_string.h"
+#include "user_interface.h"
+#include "driver/uart.h"
+#include "gpio.h"
+
+// Hardware specific
+static const uint16_t CAL_MARGIN = 20;
+static const uint8_t CTRL_LO_DFR = 0b0011;
+static const uint8_t CTRL_LO_SER = 0b0100;
+static const uint8_t CTRL_HI_X = 0b1001  << 4;
+static const uint8_t CTRL_HI_Y = 0b1101  << 4;
+static const uint16_t ADC_MAX = 0x0fff;  // 12 bits
+
+// Runtime variables
+uint16_t _width, _height;
+uint8_t _cs_pin, _irq_pin;
+int32_t _cal_dx, _cal_dy, _cal_dvi, _cal_dvj;
+uint16_t _cal_vi1, _cal_vj1;
+
+// Average pair with least distance between each
+static int16_t ICACHE_FLASH_ATTR besttwoavg( int16_t x , int16_t y , int16_t z ) {
```

Comment:
`ICACHE_FLASH_ATTR` is redundant since the default mapping for nodemcu firmware is flash. Please remove these attributes.